### PR TITLE
Add support for cluster and node pool labels

### LINKED ISSUE #101: Add support for creating and deleting node pool labels
#### Why should this feature be added?

We now support adding labels to node pools on stage. Let's add support for this to `csctl`.

#### How would you like the feature to work?

Labels can't be added until after a node pool is created. For this reason, I don't think that this work is impacted by the potential restructuring of `csctl create node-pool` in https://github.com/containership/csctl/issues/100. 

I think the following UX probably makes sense - input welcome!

Space-delimited list of labels to create (varargs):
`csctl create node-pool-label --cluster <id> --node-pool <id> labels.containership.io/key1=value1 labels.containership.io/key2=value2 ...`

Labels are first-class citizens in Containership, so they have their own IDs after creation. For that reason, the initial delete command can look just like any other delete command:

`csctl delete node-pool-label <id1> <id2> ...`

If the `labels.containership.io/` prefix is not supplied, `csctl` should automatically prepend it. This should be made extremely clear in the usage info. (I don't really love this, so I'd love to hear other people's thoughts.)

#### What alternatives have been considered?

It would be nice to also support specifying the key/value of the label for deletion, where only the key is required:

`csctl delete node-pool-label key1=-value1 key2 key3 ...`

This is not necessay for a first pass.

#### Additional Context
N/A

### PR DESCRIPTION
**Pull request by mattkelly**
#### What does this pull request accomplish?

Add support for cluster and node pool labels

 #### What issue(s) does this fix?
* Closes #101

 #### Additional Considerations

 ### Testing

 #### Setup

 `make install`

 #### Instructions

```
csctl create node-pool-labels --cluster 74f4535c-f8b0-45e8-b8fc-9d1f9960aab4 --node-pool dc4e4386-86b5-4a1b-a78f-edca91a87539 key1= key2=value2 nodepool.containership.io/key3= nodepool.containership.io/key4=value4
```
```
Label 02366d7e-3df0-47f3-8608-ecad6e561945 (key1=) created successfully
Label b6cceae9-3d42-4046-81a0-b7db2c7b3277 (key2=value2) created successfully
Label 1c4eca73-67da-4a6b-a986-c7617cc2b660 (nodepool.containership.io/key3=) created successfully
Label b0ab8a37-7ee2-4fa0-8d25-f45a8eb1df10 (nodepool.containership.io/key4=value4) created successfully
```

```
csctl get node-pool-labels --cluster 74f4535c-f8b0-45e8-b8fc-9d1f9960aab4 --node-pool dc4e4386-86b5-4a1b-a78f-edca91a87539
```
```
ID                                      LABEL                                   CREATED_AT                      UPDATED_AT
02366d7e-3df0-47f3-8608-ecad6e561945    nodepool.containership.io/key1=         2019-03-28 10:54:08 -0400 EDT   2019-03-28 10:54:08 -0400 EDT
b6cceae9-3d42-4046-81a0-b7db2c7b3277    nodepool.containership.io/key2=value2   2019-03-28 10:54:09 -0400 EDT   2019-03-28 10:54:09 -0400 EDT
1c4eca73-67da-4a6b-a986-c7617cc2b660    nodepool.containership.io/key3=         2019-03-28 10:54:09 -0400 EDT   2019-03-28 10:54:09 -0400 EDT
b0ab8a37-7ee2-4fa0-8d25-f45a8eb1df10    nodepool.containership.io/key4=value4   2019-03-28 10:54:09 -0400 EDT   2019-03-28 10:54:09 -0400 EDT
```

```
csctl delete node-pool-label --cluster 74f4535c-f8b0-45e8-b8fc-9d1f9960aab4 --node-pool dc4e4386-86b5-4a1b-a78f-edca91a87539 02366d7e-3df0-47f3-8608-ecad6e561945 b6cceae9-3d42-4046-81a0-b7db2c7b3277 1c4eca73-67da-4a6b-a986-c7617cc2b660 b0ab8a37-7ee2-4fa0-8d25-f45a8eb1df10
```
```
Node pool label 02366d7e-3df0-47f3-8608-ecad6e561945 deleted successfully
Node pool label b6cceae9-3d42-4046-81a0-b7db2c7b3277 deleted successfully
Node pool label 1c4eca73-67da-4a6b-a986-c7617cc2b660 deleted successfully
Node pool label b0ab8a37-7ee2-4fa0-8d25-f45a8eb1df10 deleted successfully
```

 ### Dependencies
* None

### CODE REVIEW

File: resource/label/label.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+	"github.com/pkg/errors"
+)
+
+// NodePoolPrefix is the prefix required for any user-defined node pool labels
+const NodePoolPrefix = "nodepool.containership.io"
+
+// ParseString parses a string of the form key=value and returns
+// the key and value or an error.
+func ParseString(labelStr string) (string, string, error) {
```
[the remark this answers is not in the document]
Comment (mattkelly) [reply resource/label/label.go]: Looks like a fixmie bug - will file a ticket.

File: resource/label/label.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	if key == "" {
+		return "", "", errors.New("key cannot be empty")
+	}
+
+	return key, val, nil
+}
+
+// String returns a string representation of a label with the given
+// key and value or an error.
+func String(key string, value string) (string, error) {
```
[the remark this answers is not in the document]
Comment (mattkelly) [reply resource/label/label.go]: This one isn't a bug, but not really a style I'd prefer to move to

File: resource/label/label.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattkelly) [reply resource/label/label.go]: Bug already exists: https://github.com/fixmie/feedback/issues/9

File: resource/label/label.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fatih) [reply resource/label/label.go]: You can disable this particular fixer, see the docs for more information: https://github.com/fixmie/docs#configuration

File: cmd/get_node_pool_label.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		}
+
+		if err != nil {
+			return err
+		}
+
+		labels := resource.NewNodePoolLabels(resp)
+
+		if len(args) == 1 {
+			resource.NodePoolLabel().DisableListView()
```
Comment (mattkelly) [inline cmd/get_node_pool_label.go]: Note that this does not work as intended - see https://github.com/containership/csctl/issues/115. I don't have a fix offhand, so I'm leaving it for consistency so it's easier to fix it everywhere in one shot (pattern recognition ⭐️).

File: resource/label/label.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattkelly) [reply resource/label/label.go]: I didn't realize it was configurable now. Thanks @fatih!

**Comment by mattkelly** [comment]
Based on our discussion around cluster labels in Slack today, I'm just going to add support for those to this PR. Please hold ☎️

**Comment by mattkelly** [comment]
Added cluster label support - this is good for a review now @containership/core-services

**Pull request merged**